notes

```diff
@@ -118,7 +118,13 @@ new_note() {
          note_name="$(generate_name)"
     fi
     mkdir -p "$(dirname "$notes_dir/$note_name")"
-    open_note "$note_name.$NOTES_EXT"
+
+    ext="${note_name#*.}"
+    if [[ ext = $note_name ]]; then # lacks extension
+        open_note "$note_name.$NOTES_EXT"
+    else
+        open_note "$note_name"
+    fi
```

Comment:
I think this will only ever add an extension if there's no `.` in the name - not if the name already has the correct extension.

I can imagine (weird) people might write notes like `mynote.2018-03-14`, and expect that to get a `.md` appended like anything else. We _could_ catch only the case where it already ends with the correct extension, but allow `.` anywhere else. Is there a specific reason you've gone down this road instead?

Reply:
thanks for pointing it out, I'll consider those case. 

> Is there a specific reason you've gone down this road instead?
Not really, i took the most naive(and obvious) approach to this problem.

P.S:
I noticed an obvious error(ext should be $ext).